core/types/celo_transaction_signing_forks_test.go

```diff
@@ -0,0 +1,100 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test_forks_activeForks tests that the correct forks are returned for a given block time and chain config
+func Test_forks_activeForks(t *testing.T) {
+	t.Parallel()
+
+	cel2Time := uint64(1000)
+
+	t.Run("Non-Celo", func(t *testing.T) {
+		config := &params.ChainConfig{
+			Cel2Time: nil,
+		}
+		assert.Equal(t, []fork(nil), celoForks.activeForks(1000, config))
+	})
+
+	t.Run("Celo1", func(t *testing.T) {
+		config := &params.ChainConfig{
+			Cel2Time: &cel2Time,
+		}
+		assert.Equal(t, []fork{&celoLegacy{}}, celoForks.activeForks(500, config))
+	})
+
+	t.Run("Celo2", func(t *testing.T) {
+		config := &params.ChainConfig{
+			Cel2Time: &cel2Time,
+		}
+		assert.Equal(t, []fork{&cel2{}, &celoLegacy{}}, celoForks.activeForks(1000, config))
+	})
+}
+
+// Test_forks_findTxFuncs tests that the correct txFuncs are returned for a given transaction type and chain config
+func Test_forks_findTxFuncs(t *testing.T) {
+	t.Parallel()
```

Comment:
Hey @Kourin1996 I think this test would be more useful if it tested from the level of the signer using a similar approach to `TestProtectedCeloLegacyTxSigning`. 

So generating a signer for a given config and then signing each transaction with that signer and checking that the sender address can be correctly recovered, of course for some transactions an error will be returned because they are not supported or deprecated.

Reply:
@piersy 

I believe such cases are covered by `TestCeloSigner_SignAndRecovery`. `TestCeloSigner_SignAndRecovery` test focuses on signing only in Celo2 but includes actual scenarios.

On the other hand, this `Test_forks_findTxFuncs` test is just a low-level test designed to detect unexpected changes.

Reply:
Good point, what is tested here is already covered by the two tests that you mentioned. It doesn't seem to me that this test is adding much value, so my preference would be to remove it, but I'm also OK with keeping it if you feel strongly about it.

Reply:
@piersy OK. I removed it!